plugins/modules/net_tools/nios/nios_host_record.py

```diff
@@ -287,17 +307,17 @@ def main():
     ''' Main entry point for module execution
     '''
     ipv4addr_spec = dict(
-        ipv4addr=dict(required=True, aliases=['address'], ib_req=True),
-        configure_for_dhcp=dict(type='bool', required=False, aliases=['dhcp'], ib_req=True),
-        mac=dict(required=False, ib_req=True),
+        ipv4addr=dict(required=True, aliases=['address']),
+        configure_for_dhcp=dict(type='bool', required=False, aliases=['dhcp']),
+        mac=dict(required=False),
```

Comment:
These are potentially behavior changing changes. You should extend `normalize_ib_spec` to recursively remove these from `options` if present, and keep them here (similar for other modules where you removed these values).

Reply:
Hi @felixfontein Thanks for reviewing! 

I had that in mind - the **only** place where this ``ib_req`` is ever used is:

https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/net_tools/nios/api.py#L242

(within ``WapiModule.run()``) and it does not care about ``suboptions`` at all. All the calls to that method pass only the "top level ``ib_spec``" variable to it (I was really glad that this nasty hack didn't go as further as using ``module.params`` for that).

```
$ git grep -nE 'run\(' plugins/module_utils/net_tools/nios/api.py plugins/modules/net_tools/nios/*.py
plugins/module_utils/net_tools/nios/api.py:237:    def run(self, ib_obj_type, ib_spec):
plugins/modules/net_tools/nios/nios_a_record.py:168:    result = wapi.run(NIOS_A_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_aaaa_record.py:155:    result = wapi.run(NIOS_AAAA_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_cname_record.py:144:    result = wapi.run(NIOS_CNAME_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_dns_view.py:139:    result = wapi.run(NIOS_DNS_VIEW, ib_spec)
plugins/modules/net_tools/nios/nios_fixed_address.py:290:    result = wapi.run(fixed_address_ip_type, ib_spec)
plugins/modules/net_tools/nios/nios_host_record.py:350:    result = wapi.run(NIOS_HOST_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_member.py:564:    result = wapi.run(NIOS_MEMBER, ib_spec)
plugins/modules/net_tools/nios/nios_mx_record.py:152:    result = wapi.run(NIOS_MX_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_naptr_record.py:189:    result = wapi.run('record:naptr', ib_spec)
plugins/modules/net_tools/nios/nios_network.py:323:    result = wapi.run(network_type, ib_spec)
plugins/modules/net_tools/nios/nios_network_view.py:127:    result = wapi.run(NIOS_NETWORK_VIEW, ib_spec)
plugins/modules/net_tools/nios/nios_nsgroup.py:436:    result = wapi.run(NIOS_NSGROUP, ib_spec)
plugins/modules/net_tools/nios/nios_ptr_record.py:157:    result = wapi.run(NIOS_PTR_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_srv_record.py:166:    result = wapi.run(NIOS_SRV_RECORD, ib_spec)
plugins/modules/net_tools/nios/nios_txt_record.py:129:    result = wapi.run('record:txt', ib_spec)
plugins/modules/net_tools/nios/nios_zone.py:230:    result = wapi.run(NIOS_ZONE, ib_spec)
```

On the chance this might be something passed as-is to the underlying ``infoblox_client.Connector`` I also grep'd that component's code for it, and it is nowhere to be found.

Similarly for the pseudo-atributes ``transform`` and ``update``, which are only used, respectively at:

* https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/net_tools/nios/api.py#L248-L249
  * within ``WapiModule.run()``
* https://github.com/ansible-collections/community.general/blob/main/plugins/module_utils/net_tools/nios/api.py#L587
  * This one from ``WapiModule.on_update()`` method, but that method is only called at:
```
plugins/module_utils/net_tools/nios/api.py:334:                    proposed_object = self.on_update(proposed_object, ib_spec)
plugins/module_utils/net_tools/nios/api.py:345:                    proposed_object = self.on_update(proposed_object, ib_spec)
plugins/module_utils/net_tools/nios/api.py:354:                    proposed_object = self.on_update(proposed_object, ib_spec)
```
from within ``WapiModule.run()``.

So it looks reasonably well-contained to me. What do you think?

Reply:
Looks reasonable, indeed.